src/wallet/spend.cpp

```diff
@@ -1141,6 +1129,16 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
            result.GetWaste(),
            result.GetSelectedValue());
 
+    // vouts to the payees
+    for (const auto& recipient : vecSend)
+    {
+        CTxOut txout(recipient.nAmount, GetScriptForDestination(recipient.dest));
+
+        if (IsDust(txout, wallet.chain().relayDustFee())) {
```

Comment:
nit: it makes sense to keep this check before coin selection to return early and avoid wasting time

Reply:
Yeah, I had considered this but it would require refactoring `IsDust` in a non-trivial way, since `IsDust` is used outside of the wallet and as such doesn't really know what a `CRecipient` is.

It's definitely possible, since all you really need to check for dust is the serialized size of the output and the value, but doing the refactor here blows up the scope of this PR quite a bit. I think it would be better to handle this in a separate PR, i.e. refactor `IsDust` so it can be used for both `CRecipients` and `CTxOuts`. Happy to open a follow-up PR, or at the very least leave a `TODO` comment here.

Reply:
Actually, I just realized I was over complicating this: you don't need to refactor `IsDust`, you just need to wrap it with an `IsDust` function that turns a `CRecipient` into a `CTxOut`, same as `GetSerializeSizeForRecipient`. Will update.

Reply:
Update, thanks @S3RK !